src/components/dashboard-widgets/dashboard-enum-widget/dashboard-enum-widget.component.ts

```diff
@@ -0,0 +1,80 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input, Output,
+    Pipe, PipeTransform, QueryList,
+    ViewChildren
+} from '@angular/core';
+import { PredefinedWidgetConfig } from '../interfaces/predefined-widget.interface';
+import { EnumConfig, EnumWidgetConfig } from '../interfaces/enum-widget.interface';
+import {coerceBooleanProperty, coerceNumberProperty} from '@angular/cdk/coercion';
+
+@Component({
+    selector: 'ux-dashboard-enum-widget',
+    templateUrl: './dashboard-enum-widget.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DashboardEnumWidgetComponent implements PredefinedWidgetConfig, EnumWidgetConfig {
+    static ngAcceptInputType_fixedMode: boolean | string;
+    static ngAcceptInputType_colSpan: number | string;
+    static ngAcceptInputType_rowSpan: number | string;
+
+    private _fixedMode: boolean = false;
+    private _colSpan: number = 1;
+    private _rowSpan: number = 1;
+
+    @Input() get fixedMode(): boolean {
+        return this._fixedMode;
+    }
+    set fixedMode(fixedMode: boolean) {
+        this._fixedMode = coerceBooleanProperty(fixedMode);
+    }
+
+    @Input() get colSpan(): number {
+        return this._colSpan;
+    }
+    set colSpan(colSpan: number) {
+        this._colSpan = coerceNumberProperty(colSpan);
+    }
+
+    @Input() get rowSpan(): number {
+        return this._rowSpan;
+    }
+    set rowSpan(rowSpan: number) {
+        this._rowSpan = coerceNumberProperty(rowSpan);
+    }
+
+    @Input() id: string = '';
+    @Input() name: string = '';
+    @Input() heading: string = '';
+
+    @Input() enums: EnumConfig[];
```

Comment:
Should be `enum` singular since it's only one set of mutually exclusive values. In truth I would prefer not using the term enum at all, probably use "select" instead of "enum".

Reply:
This is the list of possible enumeration objects, each containing value, label, and optionally icon. I was not sure if "select" would express that, so I changed it to "options".

Reply:
This is fine. Could you consider renaming the component itself as well? Enum is 1) an abbreviation, and 2) quite esoteric unless you are a programmer. We prefer to use terms which are familiar to the UX design world if at all possible.

Reply:
I renamed it to "select widget".